pkg/api/roles.go

```diff
@@ -50,157 +50,189 @@ var (
 			},
-			Grants: []string{string(models.ROLE_VIEWER)},
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     1,
-				Name:        "fixed:current:org:writer",
-				Description: "Read current organization, its quotas, and its preferences. Write current organization and its preferences.",
-				Permissions: []accesscontrol.Permission{
-					{
-						Action: ActionOrgsRead,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsQuotasRead,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsPreferencesRead,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsWrite,
-						Scope:  ScopeOrgCurrent,
-					},
-					{
-						Action: ActionOrgsPreferencesWrite,
-						Scope:  ScopeOrgCurrent,
-					},
+		Grants: []string{string(models.ROLE_VIEWER)},
+	}
+
+	currentOrgReaderRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
+			Name:        "fixed:current.org:reader",
+			DisplayName: "Current Organization reader",
+			Description: "Read the current organization, such as its ID, name, address, or quotas.",
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: ActionOrgsRead,
+					Scope:  ScopeOrgCurrent,
+				},
+				{
+					Action: ActionOrgsQuotasRead,
+					Scope:  ScopeOrgCurrent,
 				},
 			},
-			Grants: []string{string(models.ROLE_ADMIN)},
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     1,
-				Name:        "fixed:orgs:writer",
-				Description: "Create, read, write, or delete an organization. Read or write an organization's quotas.",
-				Permissions: []accesscontrol.Permission{
-					{Action: ActionOrgsCreate},
-					{
-						Action: ActionOrgsRead,
-						Scope:  ScopeOrgsAll,
-					},
-					{
-						Action: ActionOrgsWrite,
-						Scope:  ScopeOrgsAll,
-					},
-					{
-						Action: ActionOrgsDelete,
-						Scope:  ScopeOrgsAll,
-					},
-					{
-						Action: ActionOrgsQuotasRead,
-						Scope:  ScopeOrgsAll,
-					},
-					{
-						Action: ActionOrgsQuotasWrite,
-						Scope:  ScopeOrgsAll,
-					},
+		Grants: []string{string(models.ROLE_VIEWER)},
+	}
+
+	currentOrgWriterRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
+			Name:        "fixed:current.org:writer",
+			DisplayName: "Current Organization writer",
+			Description: "Read the current organization, its quotas, or its preferences. Update the current organization properties, or its preferences.",
+			Permissions: accesscontrol.ConcatPermissions(currentOrgReaderRole.Role.Permissions, []accesscontrol.Permission{
+				{
+					Action: ActionOrgsPreferencesRead,
+					Scope:  ScopeOrgCurrent,
+				},
+				{
+					Action: ActionOrgsWrite,
+					Scope:  ScopeOrgCurrent,
+				},
+				{
+					Action: ActionOrgsPreferencesWrite,
+					Scope:  ScopeOrgCurrent,
+				},
+			}),
+		},
+		Grants: []string{string(models.ROLE_ADMIN)},
+	}
+
+	orgReaderRole := accesscontrol.RoleRegistration{
```

Comment:
Not a huge fan of creating this role given front-end is not handling global assignments as of now. The writer role being complete, "overrides" local assignments and "hides" this.
Could we maybe not create it now but later, once we figure out a nice way for the front-end to work with these global permissions?